pkg/client/client.go

```diff
@@ -156,15 +156,18 @@ func NewClient(ctx context.Context, wallet *wallet.Wallet, opts ...NewClientOpti
 		return nil, fmt.Errorf("getting contract backend: %v", err)
 	}
 
-	tblContract, err := ethereum.NewClient(
-		contractBackend,
-		tableland.ChainID(config.chain.ID),
-		config.chain.ContractAddr,
-		wallet,
-		impl.NewSimpleTracker(wallet, contractBackend),
-	)
-	if err != nil {
-		return nil, fmt.Errorf("creating contract client: %v", err)
+	var tblContract *ethereum.Client
+	if contractBackend != nil {
+		tblContract, err = ethereum.NewClient(
+			contractBackend,
+			tableland.ChainID(config.chain.ID),
+			config.chain.ContractAddr,
+			wallet,
+			impl.NewSimpleTracker(wallet, contractBackend),
+		)
+		if err != nil {
+			return nil, fmt.Errorf("creating contract client: %v", err)
+		}
```

Comment:
Integrating the client into the healthbot was a good exercise... It made me realize there are times when you want to use the client without a "provider" (alchemy, infura, etc) to make contract calls. This would be cases, like the healthbot, where you're just making read calls and relayed write calls. So now the client can deal with not having a provider/contract backend supplied.